tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java

```diff
@@ -0,0 +1,239 @@
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.core;
+
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.project.core.util.ProjectUtil;
+import com.liferay.ide.sdk.core.SDK;
+import com.liferay.ide.sdk.core.SDKUtil;
+
+import java.io.File;
+import java.util.regex.Pattern;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IResourceChangeEvent;
+import org.eclipse.core.resources.IResourceChangeListener;
+import org.eclipse.core.resources.IResourceDelta;
+import org.eclipse.core.resources.IResourceDeltaVisitor;
+import org.eclipse.core.resources.WorkspaceJob;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jdt.core.ClasspathContainerInitializer;
+import org.eclipse.jdt.core.IClasspathContainer;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaCore;
+
+/**
+ * @author Simon Jiang
+ */
+public class SDKBuildPropertiesResourceListener implements IResourceChangeListener, IResourceDeltaVisitor
+{
+
+    private static final String ID_SDK_PROPERTIES_INVALID = "sdk-properties-invalid";
+
+
+    private final static Pattern buildPropertiesPattern = Pattern.compile("build.[\\w|\\W.]*properties");
+
+    protected void processPropertiesFile( IFile buildPropertiesFile ) throws CoreException
+    {
+
+        SDK sdk = SDKUtil.getWorkspaceSDK();
+
+        if ( sdk != null )
+        {
+            IProject sdkProject = SDKUtil.getWorkspaceSDKProject();
+
+            ProjectUtil.clearMarkers(sdkProject, ID_SDK_PROPERTIES_INVALID, IMarker.PROBLEM);
+
+            IStatus status = sdk.validate(true, false);
+
+            if ( !status.isOK() )
+            {
+                IStatus[] statuses = status.getChildren();
+
+                for( IStatus iStatus : statuses )
+                {
+                    ProjectUtil.setMarker(sdkProject, IMarker.PROBLEM, IMarker.SEVERITY_ERROR,
+                        iStatus.getMessage(), sdkProject.getFullPath().toPortableString(),ID_SDK_PROPERTIES_INVALID);
+                }
+
+                return;
+            }
+            sdk.getBuildProperties( true, true );
+        }
+
+        for( final IProject project : CoreUtil.getAllProjects() )
+        {
+            if ( !CoreUtil.isLiferayProject ( project ) )
```

Comment:
Should this be for only SDK projects and liferay projects instead of "all liferay" projects which would include Maven?